### PR TITLE
chore(ui-react-native): Refactor unit tests

### PR DESCRIPTION
**Pull request by ioanabrooks**
#### Description of changes

This change refactors existing React Native tests to use @testing-library instead of react-test-renderer. Also adds some assertions and tests to improve coverage.

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-native/src/InAppMessaging/components/BannerMessage/__tests__/BannerMessage.spec.tsx
Comment (calebpollman) [inline packages/react-native/src/InAppMessaging/components/BannerMessage/__tests__/BannerMessage.spec.tsx:100]: Prefer explicit assertions in unit tests over iterative assertions which require the reader to spend more time understanding the unit test. Looks like the only thing we are asserting against here is `children` in `expectedProps` so should be fairly straightforward to make the assertion explicit?

File: packages/react-native/src/InAppMessaging/components/BannerMessage/__tests__/BannerMessage.spec.tsx
Comment (calebpollman) [inline packages/react-native/src/InAppMessaging/components/BannerMessage/__tests__/BannerMessage.spec.tsx:119]: Not sure how this got past linting since it we have a rule preventing reuse of titles (`jest/no-identical-title`, probably because the title is computed), but this could be confusing for future maintainers since we are using the same title template on line 85.

File: packages/react-native/src/InAppMessaging/components/CarouselMessage/__tests__/CarouselMessage.spec.tsx
Comment (calebpollman) [inline packages/react-native/src/InAppMessaging/components/CarouselMessage/__tests__/CarouselMessage.spec.tsx:7]: If we are updating the unit tests to no longer use `react-test-renderer` directly than i would assume we would stop using the types as well?

File: packages/react-native/src/InAppMessaging/components/ModalMessage/__tests__/ModalMessage.spec.tsx
Comment (calebpollman) [inline packages/react-native/src/InAppMessaging/components/ModalMessage/__tests__/ModalMessage.spec.tsx:102]: Same comments as what i added in _BannerMessage.spec.tsx_

File: packages/react-native/src/primitives/Carousel/__tests__/Carousel.spec.tsx
Comment (calebpollman) [inline packages/react-native/src/primitives/Carousel/__tests__/Carousel.spec.tsx:13]: Same comment abt not using _react-test-renderer_ types as noted earlier

File: packages/react-native/src/primitives/Carousel/__tests__/CarouselPageIndicator.spec.tsx
Comment (calebpollman) [inline packages/react-native/src/primitives/Carousel/__tests__/CarouselPageIndicator.spec.tsx:5]: Same comment regarding _react-test-renderer_ as previously noted

File: packages/react-native/src/primitives/Carousel/types.ts
Comment (calebpollman) [inline packages/react-native/src/primitives/Carousel/types.ts:10]: Shouldn't need the `undefined` if it's optional (afaik)?

**Review by calebpollman — COMMENTED** [review]
Thanks for doing this @ioanabrooks. Left some feedback but LGTM once addressed

File: packages/react-native/src/InAppMessaging/components/CarouselMessage/__tests__/CarouselMessage.spec.tsx
Comment (ioanabrooks) [reply packages/react-native/src/InAppMessaging/components/CarouselMessage/__tests__/CarouselMessage.spec.tsx:7]: `react-native-testing-library` is built on top of react-test-renderer thus has a dependency on it, as noted in their [readme](https://github.com/callstack/react-native-testing-library#installation) as well.

**Review by calebpollman — APPROVED** [review]

**Review by joebuono — APPROVED** [review]
Thank you for doing this refactor!

**Pull request merged**
